blocks/main_loop.py

```diff
@@ -233,8 +234,20 @@ def _run_iteration(self):
         return True
 
     def _run_extensions(self, method_name, *args):
+        class Callback(str):
+            def __eq__(self, other):
+                callbackname = [key for key, value
+                                in TrainingExtension.__dict__.items()
+                                if getattr(value, '_is_callback', False)]
+                if other not in callbackname:
+                    if isinstance(other, str):
```

Comment:
got it. So now I just use TypeError for all possible errors. Like:
if other not in callback_names:
raise TypeError("{} is not a valid callback.".format(other))
return str(self) == other
And I will close this pr and make a new one.